data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/CosmosSerde.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.cosmos.operations;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.type.Argument;
+import io.micronaut.data.cosmos.common.Constants;
+import io.micronaut.data.exceptions.DataAccessException;
+import io.micronaut.data.model.runtime.RuntimePersistentEntity;
+import io.micronaut.data.model.runtime.RuntimePersistentProperty;
+import io.micronaut.jackson.core.tree.JsonNodeTreeCodec;
+import io.micronaut.json.tree.JsonNode;
+import io.micronaut.serde.Decoder;
+import io.micronaut.serde.Deserializer;
+import io.micronaut.serde.SerdeRegistry;
+import io.micronaut.serde.Serializer;
+import io.micronaut.serde.jackson.JacksonDecoder;
+import io.micronaut.serde.support.util.JsonNodeEncoder;
+import jakarta.inject.Singleton;
+
+import java.io.IOException;
+
+/**
+ * Serialize and deserialize Cosmos documents.
+ *
+ * @author radovanradic
+ * @since 4.0.0
+ */
+@Singleton
+@Internal
+public final class CosmosSerde {
```

Comment:
Does it needs to be public?

Reply:
No, changed in https://github.com/micronaut-projects/micronaut-data/pull/1813/commits/3d0e36c860c47cf909208ab8f2573f77e38465f6